### PR TITLE
Add c# 12 test project

### PR DESCRIPTION
**Pull request by bjornhellander**
I took the liberty of adding a test project for c# 12. I assume that this will have to wait until your test changes are done @sharwell , but this can be rebased or re-done easily. It was mostly mechanical work anyway. 4 (?) test cases failed on c# 12, so I updated those.

### CODE REVIEW

File: StyleCop.Analyzers/StyleCop.Analyzers.Test/ReadabilityRules/SA1130UnitTests.cs [hunk 1]
```diff
@@ -1010,5 +983,48 @@ public void Test()
...
+                Diagnostic().WithLocation(10, 25),
+                Diagnostic().WithLocation(11, 30),
+                Diagnostic().WithLocation(12, 30),
+                Diagnostic(CS1065).WithLocation(12, 53),
+                Diagnostic().WithLocation(13, 30),
+                Diagnostic(CS7014).WithLocation(13, 47),
+                Diagnostic().WithLocation(14, 30),
+                Diagnostic(CS1670).WithLocation(14, 47),
+                Diagnostic().WithLocation(15, 25),
+                Diagnostic(CS1669).WithLocation(15, 42),
```
Comment (sharwell) [inline StyleCop.Analyzers/StyleCop.Analyzers.Test/ReadabilityRules/SA1130UnitTests.cs:1002]: 💡 Let's move all the `Diagnostic()` items back into the test itself, and only have the `CS*` diagnostics in this virtual method.

File: StyleCop.Analyzers/StyleCop.Analyzers.Test/ReadabilityRules/SA1130UnitTests.cs [hunk 2]
```diff
@@ -1010,5 +983,48 @@ public void Test()
...
+            return new[]
+            {
+                Diagnostic().WithLocation(12, 30),
+                Diagnostic(CS1065).WithLocation(12, 53),
+                Diagnostic().WithLocation(13, 30),
+                Diagnostic(CS7014).WithLocation(13, 47),
+                Diagnostic().WithLocation(14, 30),
+                Diagnostic(CS1670).WithLocation(14, 47),
+                Diagnostic().WithLocation(15, 25),
+                Diagnostic(CS1669).WithLocation(15, 42),
```
Comment (sharwell) [inline StyleCop.Analyzers/StyleCop.Analyzers.Test/ReadabilityRules/SA1130UnitTests.cs:1017]: 💡 Same applies for `Diagnostic()` here

File: StyleCop.Analyzers/StyleCop.Analyzers.Test/ReadabilityRules/SA1130UnitTests.cs [hunk 3]
```diff
@@ -1010,5 +983,48 @@ public void Test()
...
+            };
+        }
+
+        protected virtual DiagnosticResult[] GetExpectedResultVerifyInvalidCodeConstructions()
+        {
+            return new[]
+            {
+                Diagnostic().WithSpan(4, 50, 4, 58),
+                DiagnosticResult.CompilerError("CS1660").WithMessage("Cannot convert anonymous method to type 'EventHandler[]' because it is not a delegate type").WithSpan(4, 50, 4, 62),
```
Comment (sharwell) [inline StyleCop.Analyzers/StyleCop.Analyzers.Test/ReadabilityRules/SA1130UnitTests.cs:1026]: 💡 Same here

File: StyleCop.Analyzers/StyleCop.Analyzers.Test/ReadabilityRules/SA1130UnitTests.cs (hunk 1 above)
Comment (bjornhellander) [reply StyleCop.Analyzers/StyleCop.Analyzers.Test/ReadabilityRules/SA1130UnitTests.cs:1002]: Sure. I thought of that too briefly, but ended up not doing it. 👍

File: StyleCop.Analyzers/StyleCop.Analyzers.Test/ReadabilityRules/SA1130UnitTests.cs (hunk 3 above)
Comment (bjornhellander) [reply StyleCop.Analyzers/StyleCop.Analyzers.Test/ReadabilityRules/SA1130UnitTests.cs]: I noticed that the only thing differing was the span, so I simplified to only check the start location. Ok simplification?

File: StyleCop.Analyzers/StyleCop.Analyzers.Test/ReadabilityRules/SA1130UnitTests.cs (hunk 1 above)
Comment (bjornhellander) [reply StyleCop.Analyzers/StyleCop.Analyzers.Test/ReadabilityRules/SA1130UnitTests.cs]: Added them as annotations in the tested code instead. This removed the need for two methods before and after the fix. Ok simplification?

**Review by sharwell — APPROVED** [review]

**Comment by sharwell** [comment]
It may be possible to split the test executions into separate jobs that run in parallel, and then merge the result in a cleanup step. For now I just hit retry on the build.

**Pull request merged**